### DISCUSSION TITLE
ScriptReference/GUILayout.VerticalScope

### DISCUSSION THREAD

**Discussion by docscomments** [Developer-Notes-Extension/Unity-Notes#192] [Notes]
# ScriptReference/GUILayout.VerticalScope

https://docs.unity3d.com/ScriptReference/GUILayout.VerticalScope.html

**Comment by mvi** [comment]
VerticalScopes are a great way to add structure and grouping to controls, especially editor inspectors and custom windows.  
In particular for editor cases using [`EditorStyles.helpBox`](https://docs.unity3d.com/ScriptReference/EditorStyles-helpBox.html) as the GUIStyle produces a well fitting scope.  
  

```csharp
GUILayout.BeginVertical(EditorStyles.helpBox);  
GUILayout.Label("Something here", EditorStyles.boldLabel);  
GUILayout.EndVertical();
```
  
Additionally you can create more interesting reusable VerticalScopes by subclassing VerticalScope, and then in the subclass adding drawing to the overridden constructor and CloseScope method. For example you could create a NamedVerticalScope class that takes a string which it displays in a nicely styled title.
